### PR TITLE
[MOB-2186] Extract ApiClient

### PR DESCRIPTION
**Pull request by vbabenkoru**

### CODE REVIEW

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java [hunk 1]
```diff
@@ -62,13 +59,20 @@
 
 //region Constructor
 //---------------------------------------------------------------------------------------
-    IterableApi() {
+IterableApi() {
+    config = new IterableConfig.Builder().build();
+}
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java:64]: indentation

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java [hunk 2]
```diff
@@ -474,23 +481,7 @@ public void track(@NonNull String eventName, int campaignId, int templateId, @Nu
             return;
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java:481]: Can `checkSDKInitialization` be also refactored somehow?
Its been called at start of many Api calls...

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java [hunk 3]
```diff
@@ -296,7 +301,9 @@ public static void initialize(@NonNull Context context, @NonNull String apiKey,
         sharedInstance.checkForDeferredDeeplink();
         IterableActivityMonitor.getInstance().registerLifecycleCallbacks(context);
         IterableActivityMonitor.getInstance().addCallback(sharedInstance.activityMonitorListener);
-        sharedInstance.inAppManager = new IterableInAppManager(sharedInstance, sharedInstance.config.inAppHandler, sharedInstance.config.inAppDisplayInterval);
+        if (sharedInstance.inAppManager == null) {
```
Comment (Ayyanchira) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java:304]: Wondering what would be a good case where `IterableApi` itself is initializing and `IterableInAppManager` wont be null? 🤔

**Comment by Ayyanchira** [comment]
Thats 400 lines gone from `IterableAPI` 👌👌👌

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java (hunk 3 above)
Comment (vbabenkoru) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java]: There is a special constructor for tests where you can inject an `inAppManager` instance.

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java (hunk 2 above)
Comment (vbabenkoru) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java:482]: Hmm, I don't think so... short of doing OOP shenanigans with an interface and 'initialized' and 'non-initialized' implementations.

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java (hunk 1 above)
Comment (vbabenkoru) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableApi.java]: Huh.. looks like an IntelliJ/Android Studio formatter bug.. It forced this indentation until I added an empty line above.

**Review by Ayyanchira — APPROVED** [review]

**Pull request merged**